app/modules/import-export/assets/js/admin.js

```diff
@@ -1,19 +1,189 @@
-			message: __( 'Removing assets or changing your site settings can drastically change the look of your website.', 'elementor' ),
+			headerMessage: __( 'Are you sure?', 'elementor' ),
+			message: __( 'Removing ', 'elementor' ) + kitNameToRemove + __( ' will permanently delete changes made to the Kit\'s content and site settings', 'elementor' ),
 			strings: {
-				confirm: __( 'Yes', 'elementor' ),
-				cancel: __( 'No, go back', 'elementor' ),
+				confirm: __( 'Delete', 'elementor' ),
+				cancel: __( 'Cancel', 'elementor' ),
 			},
-			onConfirm() {
-				location.href = revertButton.href;
+			onConfirm: this.onRevertConfirm.bind( this, event.target, kitNameToRemove ),
+		} ).show();
+	}
+
+	onRevertConfirm( revertBtn, kitNameToRemove ) {
+		const referrerKit = new URLSearchParams( revertBtn.href ).get( 'referrer_kit' );
+
+		this.setSessionStorage( Boolean( referrerKit ), referrerKit, kitNameToRemove );
+
+		location.href = revertBtn.href;
+	}
+
+	setSessionStorage( referred, referrerKit, kitNameToRemove ) {
+		sessionStorage.setItem( this.isKitReferredKey, referred.toString() );
+		if ( referred ) {
+			sessionStorage.setItem( this.kitReferrerKey, referrerKit );
+		}
+		sessionStorage.setItem( this.kitNameToRemoveKey, kitNameToRemove );
+	}
+
+	/**
+	 * MaybeShowKitReferrerDialog
+	 */
+	maybeShowKitReferrerDialog() {
+		const isKitReferred = sessionStorage.getItem( this.isKitReferredKey );
+		if ( ! isKitReferred ) {
+			return;
+		}
+
+		if ( 'false' === isKitReferred ) {
+			this.createKitDeletedWidget( {
+				message: __( 'Try a different Kit or build your site from scratch.', 'elementor' ),
+				strings: {
+					confirm: __( 'OK', 'elementor' ),
+					cancel: __( 'Kit Library', 'elementor' ),
+				},
+				onCancel: () => {
+					location.href = elementorAppConfig.base_url + '#/kit-library';
+				},
+			} );
+			this.cleanUp( false );
+
+			return;
+		}
+
+		const kitReferrerId = sessionStorage.getItem( this.kitReferrerKey );
+		this.createKitDeletedWidget( {
+			message: __( 'You\'re ready to apply a new Kit!', 'elementor' ),
+			strings: {
+				confirm: __( 'Continue to new Kit', 'elementor' ),
+				cancel: __( 'Close', 'elementor' ),
+			},
+			onConfirm: () => {
+				location.href = elementorAppConfig.base_url + '#/kit-library/preview/' + kitReferrerId;
 			},
+		} );
+		this.cleanUp( true );
+	}
+
+	/**
+	 * CreateKitDeletedWidget
+	 *
+	 * @param {Object} options
+	 */
+	createKitDeletedWidget( options ) {
+		elementorCommon.dialogsManager.createWidget( 'confirm', {
+			id: 'e-revert-kit-deleted-dialog',
+			headerMessage: sessionStorage.getItem( this.kitNameToRemoveKey ) + __( ' was successfully deleted', 'elementor' ),
+			message: options.message,
+			strings: {
+				confirm: options.strings.confirm,
+				cancel: options.strings.cancel,
+			},
+			...( options.onConfirm ) && { onConfirm: options.onConfirm },
+			...( options.onCancel ) && { onCancel: options.onCancel },
```

Comment:
```suggestion
			onConfirm: options.onConfirm ,
			onCancel: options.onCancel ,
```

You dont need the complex syntax. If the method doesnt exist it will be undefined in both ways.